### DISCUSSION TITLE
Support for MHTML/MHT Exports?

### DISCUSSION THREAD

**Discussion by Gadolinonym** [Tyrrrz/DiscordChatExporter#745] [General]
Howdy!

I understand this might not be the simplest request, but it would be admittedly MUCH easier to share an export with the media included if it was packaged as an MHTML file.

As it stands, you'd have to send somebody both the HTML file and corresponding folder (probably as a ZIP) so that they can have media load. Without it, you just have a ton of missing media/images/pfps/etc.

I suppose one clunky alternative is to have all references to media files instead be baked into the HTML file as base64 data, maybe that'd work if it was an option? 🤷‍♂️ I feel like that would be easier, even if it's scuffed.

Thank you!

**Comment by 96-LB** [comment]
With the current version you can easily share HTML files if "Download Media" is disabled since all media will use http links rather than a media folder, and the resulting export will be single-file. If they want to save images they can open it in a browser and hit ctrl-s (or right click -> save as). By saving as "webpage, complete", it'll create a media folder for them. Note that this won't catch everything exactly the same way as DiscordChatExporter, but any displayed images should be downloaded. Also, I've only done this on Chrome, so I'm not 100% other browsers work identically.

**Reply by Gadolinonym** [reply]
Hey, thanks for the update! Not 100% if I understood what you meant, so I tried opening one of the html exports where I downloaded media, opened it in Chrome, and saved the page as MHTML. When I opened up the image, the media did NOT load. I had renamed the folder of the original export to prevent it from linking back.

**Reply by 96-LB** [reply]
I wrote this a very long time ago, but if I recall correctly I was offering a potential workaround. If you export an HTML file *without* downloading media, you'll have a single HTML file. You can then share that HTML file with others and the recipient can download the media themselves by saving with the "webpage, complete" option. It's not a direct solution for MHTML exports but it makes sharing easier.
